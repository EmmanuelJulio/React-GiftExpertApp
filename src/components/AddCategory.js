import React, { useState } from "react";
import PropType from 'prop-types'

export  const AddCategory =({setCategories})=> {
  const [inputValue, setinputValue] = useState('');
  const handleInputChange = (e) => {
    setinputValue(e.target.value);
  };
  const handleSubmint = (e) => {
      e.preventDefault()
            if(inputValue.trim().length>2){
                setCategories(cats=>[inputValue,...cats])
                setinputValue('')

            }
        
  };
  return (
    <form onSubmit={handleSubmint}>
      <input
        type="text"
        value={inputValue}
        onChange={handleInputChange}
      ></input>
    </form>
  );
}

AddCategory.PropType={
    setCategories: PropType.func.isRequired
}