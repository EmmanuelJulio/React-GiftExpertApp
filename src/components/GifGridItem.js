import React from 'react'

export const GifGridItem = ( {id,title,url} ) => {
  
    
    return (
        <div className="card animate__animated  animate__backInLeft">
         <img src={url} alt={title}/>
         <p>{title}</p>
        </div>
    )
}
/*
1. Enzyme
2. Enzyme to json
3. Debe de mostrar el componente correctamente
 *shallow
 *wrapper 
 *wrapper .toMatchSnaphop().


*/