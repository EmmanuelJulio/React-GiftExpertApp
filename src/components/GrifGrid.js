import React from "react";
import { Fragment } from "react/cjs/react.production.min";
import { useFetchGifs } from "../hooks/useFetchGif";
 import { GifGridItem } from "./GifGridItem";
import PropTypes from 'prop-types';
export const GrifGrid = ({ category }) => {
  
    const {data:images,loading}= useFetchGifs( category );

  return (
    <Fragment>
    <h3 className="animate__animated  animate__fadeIn">{category}</h3>
   {loading&&<p className="animate__animated  animate__flash">Loading</p>}
    <div className="card-grid">
      
      
        {
          images.map(img=>(
            <GifGridItem
            key={img.id}
            {...img}
            />
          ))
        }
       
    </div>
    </Fragment>
  );
};
GrifGrid.propTypes={
  category:PropTypes.string.isRequired
}