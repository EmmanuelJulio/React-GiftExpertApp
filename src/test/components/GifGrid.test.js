import { shallow } from "enzyme";
import { GrifGrid } from "../../components/GrifGrid";
import React from "react";
import { useFetchGifs } from "../../hooks/useFetchGif";
import '@testing-library/jest-dom'
jest.mock('../../hooks/useFetchGif');



describe("Prueabas en <GifGrid/>", () => {
  const category = "One punch";


  test("Deeb de hacer mach con el snapshot", () => {


      useFetchGifs.mockReturnValue({
          data:[],
          loading:true
      })


    const wrapper = shallow(<GrifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });



  test('debe de mostrar cuando se cargan las imagenes UseFethGif', () => {
    const gif=[{
        id:'ABC',
        url:'https://localhost/cualquier/cosa.jpg',
        title:'cualquier cosa'

    },
    {
        id:'1234u',
        url:'https://localhost/cualquier/cosa.jpg',
        title:'cualquier cosa'

    }
  
  
  ];

    useFetchGifs.mockReturnValue({
        data:gif,
        loading:false
    })
    const wrapper = shallow(<GrifGrid category={category}/>);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe( gif.length )

   
  })
  
});
