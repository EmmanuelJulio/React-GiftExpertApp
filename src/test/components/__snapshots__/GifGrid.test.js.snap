// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueabas en <GifGrid/> Deeb de hacer mach con el snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated  animate__fadeIn"
  >
    One punch
  </h3>
  <p
    className="animate__animated  animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
